packages/strapi-admin/ee/admin/components/Roles/Permissions/ContentTypes/useFillRequiredPermissions.js

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useRef } from 'react';
+import { usePermissionsContext } from '../../../../../../admin/src/hooks';
+import {
+  contentManagerPermissionPrefix,
+  getCreateActionsSizeByContentType,
+  getAttributesByModel,
+} from '../../../../../../admin/src/components/Roles/Permissions/utils';
+
+const useFillRequiredPermissions = contentType => {
+  const { contentTypesPermissions, components, dispatch } = usePermissionsContext();
```

Comment:
I think this should be done in the reducer instead of having a hook. I'll let you decide if you think it is relevant.